src/iarc7_motion/iarc_tasks/track_roomba_task.py

```diff
@@ -91,91 +91,118 @@ def _current_velocity_callback(self, data):
+                    rospy.logerr('ObjectTrackTask: Exception when looking up transform')
+                    rospy.logerr(ex.message)
+                    return (TaskAborted(msg='Exception when looking up transform during roomba track'),)
+
+                # Creat point centered at drone's center
+                stamped_point = PointStamped()
+                stamped_point.point.x = 0
+                stamped_point.point.y = 0
+                stamped_point.point.z = 0
+
+                # returns point distances of roomba to center point of level quad
+                self._roomba_point = tf2_geometry_msgs.do_transform_point(
+                                                    stamped_point, roomba_transform)
+
+                if not self._check_max_roomba_range():
+                    return (TaskAborted(msg='The provided roomba is not found or not within a meter of the quad'),)
+
+                if self._check_max_ending_roomba_range():
+                    return (TaskDone(),)
+
+                # p-controller
+                x_vel_target = (self._roomba_point.point.x * self._K_X + 
+                            self._roomba_odometry.twist.twist.linear.x)
+                y_vel_target = (self._roomba_point.point.y * self._K_Y + 
+                            self._roomba_odometry.twist.twist.linear.y)
+                z_vel_target = self._z_holder.get_height_hold_response(
+                    self._drone_odometry.pose.pose.position.z)
+
+                #caps velocity
+                vel_target = math.sqrt(x_vel_target**2 + y_vel_target**2)
+
+                if vel_target > self._MAX_HORIZ_SPEED:
+                    x_vel_target = x_vel_target * (self._MAX_HORIZ_SPEED/vel_target)
+                    y_vel_target = y_vel_target * (self._MAX_HORIZ_SPEED/vel_target)
+                
+                if (abs(z_vel_target) > self._MAX_Z_VELOCITY):
+                    z_vel_target = z_vel_target/abs(z_vel_target) * self._MAX_Z_VELOCITY
+
+                desired_vel = [x_vel_target, y_vel_target, z_vel_target]
+
+                drone_vel_x = self._drone_odometry.twist.twist.linear.x
+                drone_vel_y = self._drone_odometry.twist.twist.linear.y
+                drone_vel_z = self._drone_odometry.twist.twist.linear.z
+
+                curr_vel = [drone_vel_x, drone_vel_y, drone_vel_z]
+
+                desired_vel = self._limiter.limit_acceleration(curr_vel, desired_vel)
+
+                velocity = TwistStamped()
+                velocity.header.frame_id = 'level_quad'
+                velocity.header.stamp = rospy.Time.now()
+                velocity.twist.linear.x = desired_vel[0]
+                velocity.twist.linear.y = desired_vel[1]
+                velocity.twist.linear.z = desired_vel[2]
+                
+                return (TaskRunning(), VelocityCommand(velocity)) 
+            else:
+                return (TaskAborted(msg='Illegal state reached in Track Roomba task'),)
```

Comment:
I think there's no need for this to be in an else block. Either way there should be a comment saying that there are no more states of the state machine to check.